core/arch/arm/mm/tee_pager.c

```diff
@@ -633,6 +576,75 @@ static vaddr_t __maybe_unused area_idx2va(struct tee_pager_area *area,
 }
 
 #ifdef CFG_PAGED_USER_TA
+static void free_area(struct tee_pager_area *area)
+{
+	tee_mm_free(tee_mm_find(&tee_mm_sec_ddr,
+				virt_to_phys(area->store)));
+	if (area->type == AREA_TYPE_RW)
+		free(area->u.rwp);
+	free(area);
+}
+
+static bool pager_add_uta_area(struct user_ta_ctx *utc, vaddr_t base,
+			       size_t size)
+{
+	struct tee_pager_area *area;
+	uint32_t flags;
+	vaddr_t b = base;
+	size_t s = size;
+
+	if (!utc->areas) {
+		utc->areas = malloc(sizeof(*utc->areas));
+		if (!utc->areas)
+			return false;
+		TAILQ_INIT(utc->areas);
+	}
+
+	flags = TEE_MATTR_PRW | TEE_MATTR_URWX;
+
+	while (s) {
+		size_t s2;
+
+		if (find_area(utc->areas, b))
```

Comment:
Just to catch an eventual conflict when `tee_pager_add_uta_area()` later is used to add areas in a more dynamic way. A conflict here would be quite tricky to find in other ways.